Search/BinarySearch.js

```diff
@@ -48,8 +48,23 @@ function binarySearchIterative (arr, x, low = 0, high = arr.length - 1) {
   // if low > high => we have searched the whole array without finding the item
   return -1
 }
+function binarySearchIterativeAlternative (arr, x) {
+  let low = -1
+  let high = arr.length
+  while (low < high - 1) {
+    const half = Math.floor((low + high) / 2)
+    // arr[half] === x => low = half else high = half
+    arr[half] < x ? (low = half) : (high = half)
+  }
+  // if arr[high] === x => return high else return -1
```

Comment:
Again unneeded comment. Please don't explain what the code does, but *why* it does it (something like: "return -1 if the element doesn't match the searched element").